### DISCUSSION TITLE
How to crop a spectrum?

### DISCUSSION THREAD

**Discussion by nicocopez** [h2020charisma/ramanchada2#60] [Q&A] [answered]
Is there any way I can crop a spectrum in ramanchada2?

**Comment by nicocopez** [comment]
I see one possibility would be to do the following (see code below). However, it only works for spe.x and spe.y, but I would get an error if I do spe[condition]. So I could go on working with spe.x and spe.y as numpy arrays, but then I would miss all the advantages of working with spectrum class.

```
## Cropping
low_end = 580
high_end = 600
condition = np.logical_and (spe.x >= low_end, spe.x < high_end)
plt.plot(spe.x[condition], spe.y[condition])
```

**Reply by georgievgeorgi** [reply]
Edit: moved as a separate answer

**Reply by nicocopez** [reply]
Thank you Georgi! That worked nicely

**Comment by georgievgeorgi** [comment, marked as answer]
Yes, we have such a method in RC2 -- [trim_axes](https://h2020charisma.github.io/ramanchada2/ramanchada2/spectrum.html#Spectrum.trim_axes)
One can specify boundaries in `'x-axis'` units or `'bin'` index via `method` parameter.

This example should do the same as yours (unless i have a typo):
```python
low_end = 580
high_end = 600
spe_cr = spe.trim_axes(method='x-axis', boundaries=(low_end, high_end))
spe_cr.plot()
```

My original answer was as a reply to the second comment. I am posting it again so that it can be marked as an answer
